horovod/common/__init__.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2016 The TensorFlow Authors. All Rights Reserved.
+# Modifications copyright (C) 2018 Uber Technologies, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+
+import ctypes
+import os
+import sysconfig
+
+
+def get_ext_suffix():
+    """Determine library extension for various versions of Python."""
+    ext_suffix = sysconfig.get_config_var('EXT_SUFFIX')
+    if ext_suffix:
+        return ext_suffix
+
+    ext_suffix = sysconfig.get_config_var('SO')
+    if ext_suffix:
+        return ext_suffix
+
+    return '.so'
+
+
+MPI_COMMON_LIB_CTYPES = \
+    ctypes.CDLL(os.path.join(os.path.dirname(__file__),
+                             'mpi_lib' + get_ext_suffix()), mode=ctypes.RTLD_GLOBAL)
+
+
+def init():
+    """A function which initializes Horovod.
+    """
+    return MPI_COMMON_LIB_CTYPES.horovod_init()
+
+
+def size():
+    """A function which returns the number of Horovod processes.
+
+    Returns:
+      An integer scalar containing the number of Horovod processes.
+    """
+    size = MPI_COMMON_LIB_CTYPES.horovod_size()
+    if size == -1:
+        raise ValueError(
+            'Horovod has not been initialized; use hvd.init().')
+    return size
+
+
+def local_size():
+    """A function which returns the number of Horovod processes within the
+    node the current process is running on.
+
+    Returns:
+      An integer scalar containing the number of local Horovod processes.
+    """
+    local_size = MPI_LIB_CTYPES.horovod_local_size()
```

Comment:
Where is this initialized in refactored code? I did not see

Reply:
It's a function in `operations.cc`